src/CLMWrapper.cpp

```diff
@@ -576,11 +570,30 @@ void ClmWrapper::callback(const sensor_msgs::ImageConstPtr& msgIn)
     active_models_st += active_m_C;
     cv::putText(disp_image, active_models_st, cv::Point(10,60), CV_FONT_HERSHEY_SIMPLEX, 0.5, CV_RGB(255,0,0));   
 
-    if(!clm_parameters[0].quiet_mode)
+    if (faceDetected)
     {
-        namedWindow("tracking_result",1);   
-        cv::imshow("tracking_result", disp_image);
+        // feeding disp_image into the cv pointer's image
+        // only if a face is detected
+        cv::cvtColor(disp_image,cv_ptr->image, CV_RGB2BGR);
     }
+     //publishing the image usign the cv pointer
+    imagePublisher.publish(cv_ptr->toImageMsg());
```

Comment:
I used cv_ptr to use cv_bridge to convert disp_image or captured_image (which were of type cv::Mat) to type Image. That's why I had the toImageMsg() function. If a face is detected, I override cv_ptr with dis_image's matrix, which contains the clm drawings. If not, I do nothing basically because cv_ptr already has the captured_image.